packages/cli/src/commands/remove-plugin/removePlugin.test.ts

```diff
@@ -161,34 +202,32 @@ describe('removePlugin', () => {
 
     describe('Removes Plugin Directory', () => {
       it('removes plugin directory from /plugins', async () => {
-        try {
-          mkTestPluginDir(testDirPath);
-          expect(fse.existsSync(testDirPath)).toBeTruthy();
-          await removePluginDirectory(testDirPath);
-          expect(fse.existsSync(testDirPath)).toBeFalsy();
-        } finally {
-          if (fse.existsSync(testDirPath)) fse.removeSync(testDirPath);
-        }
+        mkTestPluginDir(testDirPath);
+        expect(fse.existsSync(testDirPath)).toBeTruthy();
+        await removePluginDirectory(testDirPath);
+        expect(fse.existsSync(testDirPath)).toBeFalsy();
       });
     });
 
     describe('Removes System Link', () => {
       it('removes system link from @backstage', async () => {
-        const scopedDir = paths.resolveTargetRoot('node_modules', '@backstage');
-        const testSymLinkPath = path.join(
-          scopedDir,
-          `plugin-${testPluginName}`,
-        );
-        try {
-          mkTestPluginDir(testDirPath);
-          fse.ensureSymlinkSync(testSymLinkPath, testDirPath);
-
-          await removeSymLink(testSymLinkPath);
-          expect(fse.existsSync(testSymLinkPath)).toBeFalsy();
-        } finally {
-          if (fse.existsSync(testDirPath)) fse.removeSync(testDirPath);
-          if (fse.existsSync(testSymLinkPath)) fse.removeSync(testSymLinkPath);
-        }
+        const symLink = `plugin-${testPluginName}`;
+        const testSymLinkPath = `/node_modules/@backstage/${symLink}`;
+
+        mkTestPluginDir(testDirPath);
+
+        mockFs({
+          '/node_modules': {
+            '@backstage': {
+              [symLink]: mockFs.symlink({
+                path: testDirPath,
+              }),
+            },
+          },
+        });
+
+        await removeSymLink(testSymLinkPath);
```

Comment:
test truthy before deletion as well perhaps, just because i am not sure how much trust i can put on mock-fs :)